packages/lib/parse-dates.ts

```diff
@@ -9,15 +9,22 @@ import { parseZone } from "./parse-zone";
 
 type ExtraOptions = { withDefaultTimeFormat?: boolean; selectedTimeFormat?: TimeFormat };
 
-const processDate = (date: string | null | Dayjs, language: string, options?: ExtraOptions) => {
+const processDate = (
+  date: string | null | Dayjs,
+  language: string,
+  options?: ExtraOptions,
+  timeZone?: string
```

Comment:
If we really need timezone to show the correct time here, shouldn't we make it required and fix everywhere it's missing.